WooCommerce/Classes/ViewRelated/Orders/Collect Payments/CollectOrderPaymentUseCase.swift

```diff
@@ -135,13 +135,6 @@ final class CollectOrderPaymentUseCase: NSObject, CollectOrderPaymentProtocol {
                         onCancel: @escaping () -> Void,
                         onPaymentCompletion: @escaping () -> Void,
                         onCompleted: @escaping () -> Void) {
-        guard isTotalAmountValid() else {
```

Comment:
I moved it later, because it needs to happen every time a payment is retried (in case the order has changed to make the amount invalid.)

The payment retry flow doesn't cover re-connecting to the reader, so it seemed appropriate to move it to the `attemptPayment` flow which is what we retry. We now check it directly after refreshing the order, including on the first time payment is taken.

Connecting the reader is unlikely to be an unnecessary operation – for one thing, it'll often already be done (with TTP auto connection, and long lasting Bluetooth connections) and also, if they do hit this error, the reader will remain connected for another attempt at payment after they resolve the issue.

It might still be better for the merchant and customer to hit this error sooner rather than later. We need to have it in the `attemptPayment` flow too, so that it happens on retry, but we could check it twice for the first payment round. It would add a little complexity to the flow, and we may have to deal with that in future support requests, which is why I didn't leave it here, but I'm open to putting it back if you think the tradeoff is worth it.

Reply:
Thanks for the clarification, we can leave it this way 👍